replications/internal/queue_management.go

```diff
@@ -114,72 +119,73 @@ func (rq *replicationQueue) Close() error {
 }
 
+// WriteFunc is currently a placeholder for the "default" behavior
+// of the queue scanner sending data from the durable queue to a remote host.
+func WriteFunc(b []byte) error {
+	return nil
+}
+
 func (rq *replicationQueue) run() {
 	defer rq.wg.Done()
 
-	retryInterval := time.Second
-	retryTimer := time.NewTicker(retryInterval)
-	defer retryTimer.Stop()
-
-	writer := func() {
-		for {
-			_, err := rq.SendWrite(func(b []byte) error {
-				rq.logger.Info("written bytes", zap.Int("len", len(b)))
-				return nil
-			})
-			if err != nil {
-				if err == io.EOF {
-					// No more data
-					// Handle this gracefully, as it is an expected error to receive
-				} else {
-					// todo more error handling
-					panic(1)
-				}
-				break
-			}
-		}
-	}
-
 	for {
 		select {
 		case <-rq.done: // end the goroutine when done is messaged
 			return
-		case <-retryTimer.C: // run the scanner every 1s
-			writer()
 		case <-rq.receive: // run the scanner on data append
-			writer()
+			for rq.SendWrite(rq.writeFunc) {
+			}
 		}
 	}
 }
 
-func (rq *replicationQueue) SendWrite(dp func([]byte) error) (int, error) {
-	// err here can be io.EOF, indicating nothing to write
+// SendWrite processes data enqueued into the durablequeue.Queue.
+// SendWrite is responsible for processing all data in the queue at the time of calling.
+// Retryable errors should be handled and retried in the dp function.
+// Unprocessable data should be dropped in the dp function.
+// An error from this function (other than io.EOF) will crash the InfluxDB server instance.
+func (rq *replicationQueue) SendWrite(dp func([]byte) error) bool {
+
+	// Any error in creating the scanner should exit the loop in run()
+	// Either it is io.EOF indicating no data, or some other failure in making
+	// the Scanner object that we don't know how to handle.
 	scan, err := rq.queue.NewScanner()
 	if err != nil {
-		return 0, err
+		if err != io.EOF {
+			rq.logger.Error("Error creating replications queue scanner", zap.Error(err))
+		}
+		return false
 	}
 
-	var count int
 	for scan.Next() {
-		// This may return io.EOF to indicate an empty queue
-		if scan.Err() != nil {
-			err = scan.Err()
+
+		// An io.EOF error here indicates that there is no more data
+		// left to process, and is an expected error.
+		if scan.Err() == io.EOF {
 			break
 		}
-		if err = dp(scan.Bytes()); err != nil {
+
+		// Any other here indicates a problem, so we log the error and
+		// drop the data with a call to scan.Advance() later.
+		if scan.Err() != nil {
+			rq.logger.Info("Segment read error.", zap.Error(scan.Err()))
 			break
 		}
-		count += len(scan.Bytes())
-	}
 
-	if err != nil { // todo handle "skippable" errors
-		rq.logger.Info("Segment read error.", zap.Error(scan.Err()))
+		// An error here indicates an unhandlable error. Data is not corrupt, and
+		// the remote write is not retryable. This error should most likely
+		// crash out the server instance. A potential example of an error here
```

Comment:
Similar to the above, I would remove the part of the comment starting with `This error should most likely crash out...`. We shouldn't need to panic if there's an error.